--- chromium-57.0.2987.21/chrome/browser/ui/views/frame/browser_frame.cc.orig	2017-02-08 21:28:13.908935990 +0100
+++ chromium-57.0.2987.21/chrome/browser/ui/views/frame/browser_frame.cc	2017-02-08 21:27:38.568469139 +0100
@@ -42,7 +42,7 @@
 #endif
 
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS)
-#include "ui/views/widget/desktop_aura/x11_desktop_handler.h"
+//#include "ui/views/widget/desktop_aura/x11_desktop_handler.h"
 #endif
 
 ////////////////////////////////////////////////////////////////////////////////
