diff --git a/virtManager/addstorage.py b/virtManager/addstorage.py
index 1d79c4f..b9b91fb 100644
--- a/virtManager/addstorage.py
+++ b/virtManager/addstorage.py
@@ -19,7 +19,6 @@
 
 import logging
 import os
-import statvfs
 
 from gi.repository import GObject
 from gi.repository import Gtk
@@ -81,7 +80,7 @@ class vmmAddStorage(vmmGObjectUI):
 
         elif not self.conn.is_remote() and os.path.exists(path):
             vfs = os.statvfs(os.path.dirname(path))
-            avail = vfs[statvfs.F_FRSIZE] * vfs[statvfs.F_BAVAIL]
+            avail = vfs.f_frsize * vfs.f_bavail
 
         return float(avail / 1024.0 / 1024.0 / 1024.0)
 
diff --git a/virtinst/diskbackend.py b/virtinst/diskbackend.py
index 790e454..4b9eb39 100644
--- a/virtinst/diskbackend.py
+++ b/virtinst/diskbackend.py
@@ -21,7 +21,6 @@
 import logging
 import os
 import re
-import statvfs
 
 import libvirt
 
@@ -327,7 +326,7 @@ class CloneStorageCreator(_StorageCreator):
         ret = False
         msg = None
         vfs = os.statvfs(os.path.dirname(self._path))
-        avail = vfs[statvfs.F_FRSIZE] * vfs[statvfs.F_BAVAIL]
+        avail = vfs.f_frsize * vfs.f_bavail
         need = int(self._size * 1024 * 1024 * 1024)
         if need > avail:
             if self._sparse:
